flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSource.java

```diff
@@ -90,12 +94,8 @@ private IcebergTableSource(TableLoader loader, TableSchema schema, Map<String, S
 
   @Override
   public void applyProjection(int[][] projectFields) {
-    this.projectedFields = new int[projectFields.length];
-    for (int i = 0; i < projectFields.length; i++) {
-      Preconditions.checkArgument(projectFields[i].length == 1,
-          "Don't support nested projection in iceberg source now.");
-      this.projectedFields[i] = projectFields[i][0];
-    }
+    this.projectedFields = projectFields;
+    this.topProjectedFields = Arrays.stream(projectFields).mapToInt(array -> array[0]).distinct().toArray();
```

Comment:
Use topProjectedFields to extract data from the original file. In this case, the nested fields in the data remain the same.
Use projectedFields to flatten the original data. In this case, nested fields are extracted.